pygenometracks/tracks/GwasTrack.py

```diff
@@ -0,0 +1,163 @@
+    BOOLEAN_PROPERTIES = []
+    STRING_PROPERTIES = ['title', 'file_type', 'file', 'ylabel', 'y_values_format', 'color']
+    FLOAT_PROPERTIES = {'height': [0, np.inf], 'fontsize': [0, np.inf], 'id_fontsize': [0, np.inf], 'cs_dotsize': [0, np.inf], 'y_axis_max_val': [0, np.inf]}
+    INTEGER_PROPERTIES = {}
+
+    def __init__(self, *args, **kwarg):
+        super(GwasTrack, self).__init__(*args, **kwarg)
+
+    def process_gwas(self):
+        """
+        Process input .gwas file to a pandas dataframe.
+        Read the file, check it has all the required columns. Add CS and INT columns if not present.
+        If y_values_format is -log10, transform the P-values to -log10(P).
+
+        :return: pandas dataframe with the data, and the maximum value of the y-axis
+        """
+        f = self.properties['file']
+        df = pd.read_csv(f, sep='\t')
+        # Check the columns of the df include the required columns (CHR, BP, SNP, P)
+        required_columns = ['CHR', 'BP', 'SNP', 'P']
+        for column in required_columns:
+            if column not in df.columns:
+                raise ValueError(f"File {f} does not contain required column {column}")
+
+        # Add columns for CS and INT if they don't exist (standard IGV .gwas format)
+        if 'CS' not in df.columns:
+            df['CS'] = '0'  # TODO: these should be set to 0 when we make the change to 0/1 instead of NO/YES. /done now
+        if 'INT' not in df.columns:
+            df['INT'] = '0'
+
+        # For the -log10 scale, calculate the -log10 of the P-value
+        if self.properties['y_values_format'] == '-log10':
+            df['P'] = df['P'].apply(lambda v: -np.log10(v))
+
+        # Maximum value of the y-axis, rounded up
+        maxval = np.ceil(df['P'].max())
+
+        return df, maxval
+
+    def plot(self, ax, chrom, region_start, region_end):
+        """
+        Plot the given title at a fixed location in the axis.
+        The chrom, region_start and region_end variables are not used at the moment.
+
+        :param ax: matplotlib axis to plot
+        :param chrom_region: chromosome name
+        :param start_region: start coordinate of genomic position
+        :param end_region: end coordinate
+        """
+
+        df, max_y = self.process_gwas()
+        if self.properties['y_axis_max_val']:
+            max_y = self.properties['y_axis_max_val']
+
+        x = df['BP']
+
+        def floor(val):
+            """
+            Alter values to be between 0.02 and 0.95 to avoid cropping by matplotlib.
+            TODO: This is a hacky and temporary solution, and does not apply when P is a P-value and not a PP.
+            """
+            if val > 0.95:
+                return 0.95
+            elif val < max_y/20:
+                return max_y/20
+            else:
+                return val
+
+        if self.properties['y_values_format'] == 'PP':
+            y = df['P'].apply(floor)
```

Comment:
I don't know GWAS but I am under the impression that with this line all y values will be to 0, no?